Do not recommend alpine

sciurus opened the pull request:
Based on our IRC conversation, we do not want people to use the alpine variants of images.

jbuck commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
+FROM node:4.8.7
 
 # add a non-privileged user for installing and running
 # the application
```
Bah, I can't add this comment to the right line, but I bet the addgroup/adduser commands won't work here - they're different between alpine & debian

jbuck commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM node:4.8.7-alpine
+FROM node:4.8.7
```
I'd recommend changing this to `node:10`. Rather than updating it for every minor or patch version, just keep rolling forward

jbuck commented on docs/building-container.md:
```diff
@@ -6,11 +6,11 @@
  [Here](https://hub.docker.com/explore/) is a list of the officially supported docker images.
 
-* Build the smallest container you can. If you're looking to shrink the size of your Docker image, you should consider using the -slim/-alpine tags within each Official image. They ship without build tools, or use an entirely different toolchain that lets you start with a smaller base image.
+* Build the smallest container you can. If you're looking to shrink the size of your Docker image, you should consider using the -slim tag within each Official image. They ship with fewr libraries and tools, so you start with a smaller base image.
```
s/fewr/fewer/

sciurus commented:
Thanks @jbuck ; fixed.

I noticed that the old Dockerfile installed the application as the app user. This meant the app user owned the node modules and thus could write to them, counter to our own recommendation that "this user is for privilege deescalation and should have read-only access to files". I've fixed that.

I still had to make the /app directory writable by the app user, otherwise `npm run` failed with a complaint that "npm update check failed"

mostlygeek commented:
Curious, what are the problems supporting alpine containers?

sciurus commented on Dockerfile:
```diff
@@ -1,21 +1,21 @@
-    adduser -D -G app -h /app -u 10001 app
+# add a non-privileged user for running the application
+RUN groupadd --gid 10001 app && \
+    useradd -g app --uid 10001 --shell /usr/sbin/nologin --create-home --home-dir /app app
```
Don't we expect the parent image to be up to date? Quoting https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

> Avoid RUN apt-get upgrade and dist-upgrade, as many of the “essential” packages from the parent images cannot upgrade inside an unprivileged container. If a package contained in the parent image is out-of-date, contact its maintainers. If you know there is a particular package, foo, that needs to be updated, use apt-get install -y foo to update automatically.

jvehent replied:
I suppose we do. And official images are probably kept up to date.

mostlygeek commented:
@sciurus those are very good reasons not to use alpine.  The -slim packages seem like the right trade off between size and security then for us.

sciurus commented:
@jbuck take another look when you get a change.
